src/router/router.ts

```diff
@@ -69,7 +69,18 @@ export class Router {
    * @returns
    */
   private getLocalizedRoutes(locale: string) {
-    return this.schema.routes[locale] || []
+    return (
+      this.schema.routes[locale]?.sort((a, b) => {
+        const hasSlugA = a.name.indexOf('[') !== -1
+        const hasSlugB = b.name.indexOf('[') !== -1
+
+        if (hasSlugA && hasSlugB) {
+          return 0
```

Comment:
IMO we also need to sort nested dynamic routes. 

So that e.g:

```
- /articles/[id]
- /[…slug]
- /users/[id]
- /users
- /articles
```

becomes:

```
- /articles
- /users
- /articles/[id]
- /users/[id]
- /[…slug] // this must be the last
```

Instead of `return 0` I would do sth `return dynamicIndexB - dynamicIndexA`.

We can define `dynamicIndexA` and `dynamicIndexB` before line 74 and reuse it here an for calculating`isDynamicA` and `isDynamicB`

Reply:
Done